### ISSUE TITLE
(aws_ec2): Can't get association_default_route_table_id from CfnTransitGateway 

### ISSUE THREAD

**Issue by CarlosDomingues** [aws/aws-cdk#18104]
### What is the problem?

I can't  reference the `association_default_route_table_id` attribute from a `CfnTransitGateway` instance.

### Reproduction Steps

```python

[...]

# Create Transit Gateway
my_tgw= aws_ec2.CfnTransitGateway(self, "MyTgw",
    auto_accept_shared_attachments="enable"
)

# Try creating a route by refering to AssociationDefaultRouteTableId 
# ERROR: [...] Requested attribute AssociationDefaultRouteTableId must be a readonly property in schema for AWS::EC2::TransitGateway
route_table_id = my_tgw.get_att("AssociationDefaultRouteTableId").to_string()

route = aws_ec2.CfnTransitGatewayRoute(self, "Route",
    transit_gateway_route_table_id=route_table_id ,
)

```

### What did you expect to happen?

I expected to be able to reference AssociationDefaultRouteTableId without issues.

### What actually happened?

I got an error: `ERROR: [...] Requested attribute AssociationDefaultRouteTableId must be a readonly property in schema for AWS::EC2::TransitGateway`

### CDK CLI Version

2.0.0-rc.33

### Node.js Version

v16.13.0

### OS

Linux

### Language

Python

### Language Version

Python 3.8.12

**Comment by peterwoodworth** [comment]
Cfn resources are identical to CloudFormation resources, so you'll be relying on CloudFormation here.

The `getAtt` function is identical to the CloudFormation function [`Fn::GetAtt`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-getatt.html). 

You can view which attributes are retrievable from resources by going to the CloudFormation docs in the given resource's `return values` section. [Example](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpc.html#aws-resource-ec2-vpc-return-values)

So in other words, attributes are NOT equal to properties. [Id](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-transitgateway.html#aws-resource-ec2-transitgateway-return-values) is the only attribute available for transit gateway

**Comment by CarlosDomingues** [comment]
@peterwoodworth thanks for the explanation!

From the [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-transitgateway.html#aws-resource-ec2-transitgateway-return-values), it seems only the `Id` attribute is returned.

So my options here are either creating a custom resource or filling the id manually, right?

**Comment by peterwoodworth** [comment]
Yes - it seems to me that those would be the best options. I just tried some other options out - but they didn't end up working

**Comment by CarlosDomingues** [comment]
@peterwoodworth understood! I think we can close this. Again, thank you so much for the guidance! I migrated to AWS CDK from Terraform and know very little about Cloud Formation.
